sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/DataSource.scala

```diff
@@ -432,6 +433,32 @@ case class DataSource(
   }
 
   /**
+   * Return all paths represented by the wildcard string.
+   * Follow [[InMemoryFileIndex]].bulkListLeafFile and reuse the conf.
+   */
+  private def getGlobbedPaths(
```

Comment:
Could you move this method to `object DataSource`?

Reply:
Done in next commit.